src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java

```diff
@@ -121,6 +123,9 @@ private IAIState startWorking()
             getOwnBuilding().ClearCurrentGrave();
         }
 
+        // Check for totem in order to grant advancement
+        getTotemResurrectChance();
```

Comment:
Nah, a lot of unnecessary checks, it's fine to just give the advancement only on the attempt to resurrect